flocker/node/agents/functional/test_cinder.py

```diff
@@ -39,9 +39,20 @@ def cinderblockdeviceapi_for_test(test_case, cluster_id):
         by ``TidyCinderVolumeManager`` to cleanup any lingering volumes that
         are created during the course of ``test_case``
     """
+    local_instance_uuid = _instance_uuid()
+    # Create the nova client last so that its cleanup method is
+    # run first and detaches volumes before the cinder client
+    # cleanup then deletes them.
+    # Yuck!
+    cinder_client = tidy_cinder_client_for_test(test_case)
+    nova_client = tidy_nova_client_for_test(test_case)
     return cinder_api(
-        cinder_client=tidy_cinder_client_for_test(test_case),
+        cinder_client=cinder_client,
+        nova_client=nova_client,
         cluster_id=cluster_id,
+        host_map={
+            u'192.0.2.123': local_instance_uuid,
```

Comment:
Fortunately, our current tests only ever attach to a single host.
The attach_elsewhere_attached test fortunately attaches to *this* host first and then some other host, but that fails because of the early check for a `BlockDeviceVolume.host` value in `CinderBlockDeviceAPI.attach`